src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -479,6 +479,8 @@ private static CodeTypeBase ConvertComposedTypeToWrapper(CodeClass codeClass, Co
         if (!supportsInnerClasses)
         {
             var @namespace = codeClass.GetImmediateParentOfType<CodeNamespace>();
+            if (@namespace.FindChildByName<CodeClass>(codeComposedType.Name, false) is not null)
+                codeComposedType.Name = $"{codeComposedType.Name}Wrapper";
```

Comment:
```suggestion
                @namespace.RenameChildElement(codeComposedType.Name, $"{codeComposedType.Name}Wrapper");
```

Reply:
Thanks for the suggestion, but this would over-write the already existing class e.g model in the namespace. I'm trying to ensure the composed type wrapper is created with a unique name.

Reply:
This would overwrite the name of an already existing class in that namespace e.g. a model. The fix is to ensure composed type wrapper classes don't conflict with an already existing class.

Reply:
> This would overwrite the name of an already existing class in that namespace e.g. a model. The fix is to ensure composed type wrapper classes don't conflict with an already existing class.

+1 This is similar to the check in line 519 for languages that don't support inner classes.

Reply:
thanks for the clarification